EasyImgur/Form1.cs

```diff
@@ -320,15 +320,32 @@ private void UploadClipboard( bool _Anonymous )
         private void UploadAlbum( bool _Anonymous, string[] _Paths, string _AlbumTitle )
         {
             ShowBalloonTip(2000, "Hold on...", "Attempting to upload album to Imgur (this may take a while)...", ToolTipIcon.None);
-            List<Image> images = new List<Image>();
+            List<string> validImgPaths = new List<string>();
+            bool hasAtLeastOneFullyValidatedImageFile = false;
             List<string> titles = new List<string>();
             List<string> descriptions = new List<string>();
             int i = 0;
             foreach (string path in _Paths)
             {
+                // instead of loading every file into memory, peek only their header to check if they are valid images
                 try
                 {
-                    images.Add(Image.FromStream(new MemoryStream(File.ReadAllBytes(path))));
+                    if (FileHelper.CheckIfValidImageFileHeader(path))
+                        validImgPaths.Add(path);
+
+                    // in rare case the file might have valid image file header, but upon reading it throws error (maybe corrupted file)
+                    // we can skip those files but at that time code has already done create new album
+                    // so we will only create album if there is at least one fully valid image
+                    if (!hasAtLeastOneFullyValidatedImageFile)
+                    {
+                        try
+                        {
+                            Image.FromStream(new MemoryStream(File.ReadAllBytes(path)));
```

Comment:
Hmm, now there's *three* file reads practically back to back: one in `CheckIfValidImageFileHeader`, on here, then one in `ImgurAPI.UploadAlbum`. Couldn't the `FileHelper` function just load the file into an `Image` like here? Isn't that enough of a validation for us (see my first comment)?

Reply:
The reason, as specified in my reply to your first comment, is to prevent the case where album is created but later has no valid image to be uploaded (error when `Image.FromStream`).
Indeed with this approach the code will have to load extra 1 image, which can be expensive...
Perhaps delete album afterwards if no valid image has been uploaded would be a better approach?

Reply:
@bkeiren Your thoughts on album uploading?

a) Just upload the images straight to Imgur and delete the empty album afterwards if no images were valid
b) Check each file's headers to see if it seems like a valid image, only upload the valid images if there are any
c) ???

Reply:
@cxcorp @sobaarcier I would go with option A, I think. Option B sounds nice enough in theory but in the end it'll still be a best-effort approach on our end, and then if all images *seemed* valid but according to Imgur weren't actually valid, you still need to deal with that album somehow. I think it makes sense to just try to upload without doing advance checking, and if nothing succeeded then we get rid of the album again (could actually turn this into a setting too in case anyone wants to configure this).

Also, we should check whether Imgur has a specific response for invalid images in which case we can provide a convenient error message to the user to display the reason for why the upload failed.

Reply:
Hi @sobaarcier! Are you still interested in making the changes? Looks like you've found something really worth fixing! (#52)